src/transactions/OperationFrame.h

```diff
@@ -98,5 +99,8 @@ class OperationFrame
     virtual bool isDexOperation() const;
 
     virtual bool isSoroban() const;
+
+    virtual bool
+    validateSorobanOpSize(SorobanNetworkConfig const& config) const;
```

Comment:
Yeah that was why I did it this way.
We would have to pass app to both `checkValid` (this is not virtual) and `doCheckValid` (this is pure virtual so every operation implements) and will touch all existing operation types.
Doing it this way felt like the least disruptive to me. But I'm happy to change it if you feel strongly otherwise.

Reply:
Removed this function and put the logic inside `doCheckValid` as suggested.